lib/pure/selectors.nim

```diff
@@ -314,6 +314,14 @@ else:
     key.events = {}
     key.data = empty
 
+  proc verifySelectParams(timeout: int) =
+    # Timeout of -1 means: wait forever
+    # Anything higher is the time to wait in miliseconds.
+    if timeout < -1:
```

Comment:
use ``doAssert(timeout >= -1)`` instead, programming bugs are not exceptions to be handled.

Reply:
Then why do we even have a `ValueError` exception? Isn't that its purpose?

Reply:
Regex parsing can fail because the user's input might not be a valid regex --> exception, programmer cannot pass the right value to ``timeout`` --> a bug, not an exception.

Reply:
Makes sense.